libraries/rush-lib/src/logic/base/BaseInstallManager.ts

```diff
@@ -449,11 +450,22 @@ export abstract class BaseInstallManager {
         // Clear the currently installed git hooks and install fresh copies
         FileSystem.ensureEmptyFolder(hookDestination);
 
+        // Find the relative path from Git hooks directory to the directory storing the actual scripts.
+        const hookRelativePath: string = Path.convertToSlashes(path.relative(hookDestination, hookSource));
+
         // Only copy files that look like Git hook names
         const filteredHookFilenames: string[] = hookFilenames.filter((x) => /^[a-z\-]+/.test(x));
         for (const filename of filteredHookFilenames) {
-          // Copy the file.  Important: For Bash scripts, the EOL must not be CRLF.
-          const hookFileContent: string = FileSystem.readFile(path.join(hookSource, filename));
+          const hookFileContent: string = `#!/bin/bash
+SCRIPT_DIR="$( cd "$( dirname "\${BASH_SOURCE[0]}" )" &> /dev/null && pwd )"
+SCRIPT_IMPLEMENTATION_PATH="$SCRIPT_DIR/${hookRelativePath}/${filename}"
+if [[ -f "$SCRIPT_IMPLEMENTATION_PATH" ]]; then
+  exec "$SCRIPT_IMPLEMENTATION_PATH"
+else
+  echo "The ${filename} Git hook does not exist in your version of the repo. Either remove the ${filename} hook file (e.g. '.git/hooks/${filename}') in case this feature is not used anymore in main branch. Otherwise, updating your branch to current mainstream will solve the issue."
```

Comment:
```suggestion
  echo "The ${filename} Git hook no longer exists in your version of the repo. Run 'rush install' or 'rush update' to refresh your installed Git hooks."
```

Reply:
This should also probably return a nonzero exit code.

Reply:
Why would you return a nonzero exit code? 🤔

It would stop git commands for pre-* hooks but would just go away with post-* hooks. I believe it does not matter that much if these files will remain in the directory for some period of time. What are the down sides of having these files in the directory and getting this message in console?

What are the use cases to remain there for a long time? I can see the following:

1. Changing a branch without running `rush install` or `rush update`. I do not think this is much likely to happen.
2. Removing Rush.js from monorepo completely. I guess this is even less likely to happen.

Am I missing something?